openhcl/underhill_core/src/worker.rs

```diff
@@ -1876,42 +1866,44 @@ async fn new_underhill_vm(
         crate::inspect_proc::periodic_telemetry_task(driver_source.simple()),
     );
 
-    let nvme_manager = if env_cfg.nvme_vfio {
-        let shared_vis_pool_spawner = shared_vis_pages_pool
-            .as_ref()
-            .map(|p| p.allocator_spawner());
+    let shared_vis_pool_spawner = shared_vis_pages_pool
+        .as_ref()
+        .map(|p| p.allocator_spawner());
 
-        let private_pool_spanwer = private_pool.as_ref().map(|p| p.allocator_spawner());
+    let private_pool_spanwer = private_pool.as_ref().map(|p| p.allocator_spawner());
 
-        let save_restore_supported = env_cfg.nvme_keep_alive && private_pool_spanwer.is_some();
-        let vfio_dma_buffer_spawner = Box::new(
-            move |device_id: String| -> anyhow::Result<Arc<dyn VfioDmaBuffer>> {
-                shared_vis_pool_spawner
-                    .as_ref()
-                    .map(|spawner| {
-                        spawner
-                            .allocator(device_id.clone())
-                            .map(|alloc| Arc::new(alloc) as _)
-                    })
-                    .unwrap_or_else(|| {
-                        private_pool_spanwer
-                            .as_ref()
-                            .map(|spawner| {
-                                spawner
-                                    .allocator(device_id.clone())
-                                    .map(|alloc| Arc::new(alloc) as _)
-                            })
-                            .unwrap_or(Ok(Arc::new(LockedMemorySpawner) as _))
-                    })
-            },
-        );
+    let vfio_dma_buffer_spawner = Box::new(
+        move |device_id: String| -> anyhow::Result<Arc<dyn VfioDmaBuffer>> {
+            shared_vis_pool_spawner
+                .as_ref()
+                .map(|spawner| {
+                    spawner
+                        .allocator(device_id.clone())
+                        .map(|alloc| Arc::new(alloc) as _)
+                })
+                .unwrap_or_else(|| {
+                    private_pool_spanwer
+                        .as_ref()
+                        .map(|spawner| {
+                            spawner
+                                .allocator(device_id.clone())
+                                .map(|alloc| Arc::new(alloc) as _)
+                        })
+                        .unwrap_or(Ok(Arc::new(LockedMemorySpawner) as _))
+                })
+        },
+    );
+
+    let dma_manager = GlobalDmaManager::new(vfio_dma_buffer_spawner);
+    let nvme_manager = if env_cfg.nvme_vfio {
+        let save_restore_supported = env_cfg.nvme_keep_alive; //&& private_pool_spanwer.is_some();
```

Comment:
is this correct?

Reply:
I think you still need to check that the private pool is available right? you may need to keep this as a separate bool?